### ISSUE TITLE
creating fabric networks still broken

### ISSUE THREAD

**Issue by bahamat** [joyent/node-triton#255]
I'm unable to create fabric networks.

```
triton network create --name example-network --subnet 10.30.30.0/24 --start_ip=10.30.30.5 --end_ip=10.30.30.250 --resolvers=8.8.8.8 30                                                   
triton network create: error (InvalidArgument): property "resolvers": string value found, but a array is required
```

Additionally, neither `--resolvers='["8.8.8.8","8.8.4.4"]'` nor `--resolvers=8.8.8.8,8.8.4.4` work.

**Comment by jclulow** [comment]
The code for `triton network create` was clearly not tested well.  It also explodes if you don't provide options which are necessary, such as `--name`:

```
$ triton network create 40

/ws/node-triton/node_modules/assert-plus/assert.js:22
    throw new assert.AssertionError({
    ^
AssertionError: opts.name (string) is required
    at CloudApi.createFabricNetwork (/ws/node-triton/lib/cloudapi2.js:526:12)
    at onSetup (/ws/node-triton/lib/do_network/do_create.js:84:18)
    at /ws/node-triton/lib/common.js:888:13
    at promptPassphraseUnlockKey (/ws/node-triton/lib/common.js:823:9)
    at /ws/node-triton/lib/common.js:885:9
    at /ws/node-triton/lib/tritonapi.js:526:13
    at /ws/node-triton/node_modules/smartdc-auth/lib/keyring.js:193:3
    at /ws/node-triton/node_modules/smartdc-auth/lib/keyring.js:148:3
    at /ws/node-triton/node_modules/smartdc-auth/node_modules/vasync/lib/vasync.js:83:5
    at /ws/node-triton/node_modules/smartdc-auth/lib/keyring.js:159:4
```

The option names are similarly inconsistent with the rest of the tool; e.g., `--no_nat` instead of `--no-nat`.

**Comment by jclulow** [comment]
Change for review @ https://cr.joyent.us/4763

**Comment by jclulow** [comment]
@bahamat If you install the `fix_fabrics` branch, you should be able to specify multiple `--resolver (-r)` options to get DNS servers in your network; e.g., via `npm install git+https://github.com/joyent/node-triton#fix_fabrics`.

New usage information is available from `triton network create --help`:

```
Create a network on a VLAN.

Usage:
    triton network create [OPTIONS] VLAN

Options:
    -h, --help  Show this help.

  Create options:
    -n NAME, --name=NAME
                Name of the NETWORK.
    -D DESC, --description=DESC
                Description of the NETWORK.

    -s SUBNET, --subnet=SUBNET
                A CIDR string describing the NETWORK.
    -S START_IP, --start-ip=START_IP, --start_ip=START_IP
                First assignable IP address on NETWORK.
    -E END_IP, --end-ip=END_IP, --end_ip=END_IP
                Last assignable IP address on NETWORK.

    -g IP, --gateway=IP
                Default gateway IP address.
    -r RESOLVER, --resolver=RESOLVER
                DNS resolver IP address. Specify multiple -r options for
                multiple resolvers.
    -R SUBNET=IP, --route=SUBNET=IP
                Static route for network. Each route must include the subnet (IP
                address with CIDR prefix length) and the router address. Specify
                multiple -R options for multiple static routes.

    -x, --no-nat, --no_nat
                Disable creation of an Internet NAT zone on GATEWAY.

  Other options:
    -j, --json  JSON stream output.


Examples:
    Create the "accounting" network on VLAN 1000:
        triton network create -n accounting --subnet 192.168.0.0/24 \
            --start-ip 192.168.0.1 --end-ip 192.168.0.254 --no-nat \
            1000

    Create the "eng" network on VLAN 1001 with a pair of static routes:
        triton network create -n eng -s 192.168.1.0/24 \
            -S 192.168.1.1 -E 192.168.1.249 --no-nat \
            --route 10.1.1.0/24=192.168.1.50 \
            --route 10.1.2.0/24=192.168.1.100 \
            1001

    Create the "ops" network on VLAN 1002 with DNS resolvers and NAT:
        triton network create -n ops -s 192.168.2.0/24 \
            -S 192.168.2.10 -E 192.168.2.249 \
            --resolver 8.8.8.8 --resolver 8.4.4.4 \
            --gateway 192.168.2.1 \
            1002
```

Can you give that a try?

**Comment by bahamat** [comment]
It's working, this looks good to me.

**Comment by jclulow** [comment]
## Testing Notes

In addition to checking the updated `triton network create --help` output (above), I have been using these changes to manipulate networks in our staging environment for some time now.  I was able to successfully execute each of the documented examples (above), as well as other combinations of resolvers and routes that I have needed as part of setting up a test environment for Gerrit.

**Comment by jclulow** [comment]
Published to NPM registry as `triton@6.2.0`.
